interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/GlobalVariableDefinitionExtractor.scala

```diff
@@ -1,44 +1,35 @@
 package pl.touk.nussknacker.engine.definition
 
-import pl.touk.nussknacker.engine.api.MetaData
-import pl.touk.nussknacker.engine.api.definition.TypedNodeDependency
 import pl.touk.nussknacker.engine.api.process.WithCategories
 import pl.touk.nussknacker.engine.api.typed.TypedGlobalVariable
 import pl.touk.nussknacker.engine.api.typed.typing.Typed
-import pl.touk.nussknacker.engine.definition.DefinitionExtractor.{ObjectDefinition, ObjectWithMethodDef, StandardObjectWithMethodDef}
-import pl.touk.nussknacker.engine.definition.MethodDefinitionExtractor.{MethodDefinition, OrderedDependencies}
+import pl.touk.nussknacker.engine.definition.DefinitionExtractor.{ComponentImplementationInvoker, ObjectDefinition, ObjectWithMethodDef, StandardObjectWithMethodDef}
 
 object GlobalVariableDefinitionExtractor {
 
+  import pl.touk.nussknacker.engine.util.Implicits._
+
   def extractDefinitions(objs: Map[String, WithCategories[AnyRef]]): Map[String, ObjectWithMethodDef] = {
-    objs.map { case (varName, varWithCategories) =>
-      (varName, extractDefinition(varName, varWithCategories))
-    }
+    objs.mapValuesNow(extractDefinition)
   }
 
-  private def extractDefinition(varName: String, varWithCategories: WithCategories[AnyRef]): StandardObjectWithMethodDef = {
+  def extractDefinition(varWithCategories: WithCategories[AnyRef]): StandardObjectWithMethodDef = {
     val returnType = varWithCategories.value match {
       case typedGlobalVariable: TypedGlobalVariable => typedGlobalVariable.initialReturnType
       case obj => Typed.fromInstance(obj)
     }
-    val methodDef = MethodDefinition(
-      name = varName,
-      invocation = (obj, deps) => obj match {
-        case typedGlobalVariable: TypedGlobalVariable => typedGlobalVariable.value(deps.head.asInstanceOf[MetaData])
-        case _ => obj
-      },
-      orderedDependencies = new OrderedDependencies(List(TypedNodeDependency[MetaData])),
-      returnType = returnType,
-      // Used only for services.
-      runtimeClass = classOf[Any],
-      annotations = Nil
-    )
     val objectDef = ObjectDefinition(
       parameters = Nil,
       returnType = Some(returnType),
       categories = varWithCategories.categories,
       componentConfig = varWithCategories.componentConfig
     )
-    StandardObjectWithMethodDef(varWithCategories.value, methodDef, objectDef)
+    StandardObjectWithMethodDef(
+      ComponentImplementationInvoker.nullImplementationInvoker,
```

Comment:
Is it correct to pass null invoker here? Do variables ever get invoked?

Reply:
Variables are always accessed via `StandardObjectWithMethodDef.obj`, I added a comment about that